src/logic/calculate.js

```diff
@@ -0,0 +1,39 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  const newData = dataObject;
+  switch (buttonName) {
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+      newData.next += buttonName;
+      break;
+
+    case '+/-':
+      newData.total *= -1;
+      newData.next *= -1;
+      break;
+
+    case 'AC':
+      newData.total = '0';
+      newData.next = '0';
+      newData.operation = null;
+      break;
+
+    default:
+      newData.total = operate(...dataObject);
+      newData.next = '0';
+      newData.operation = null;
+      break;
+  }
+  return newData;
```

Comment:
we need to add more conditions, like;
1. division zero error handling(if we divide a digit by zero we should get an error message)
2. when `.` return `0.`